modules/ROOT/pages/apikit-odatav4-expand-feature.adoc

```diff
@@ -0,0 +1,98 @@
+This is used to select which `<entity-collection-source/>` will be executed when expanding that property.
+. Each expandable `NavigationProperty` should have a `ReferentialConstraint` attached. +
+This is used to create the inner query where the value from the `Property` will be matched against the value of the `ReferencedProperty`.
+
+=== Flow Execution
+
+. The main flow is executed and completed resulting in a collection of `entities`.
+. The OData module gathers all the properties that need expansion.
+. For each `NavigationProperty`:
+.. Get the `source property` from its `ReferentialConstraint`.
+.. Get the `referenced property` from its `ReferentialConstraint`.
+.. Get the `destination entity set` from its `NavigationPropertyBinding`.
+.. For each of the `entities` get the resolved `source property`. Put all these values on a `source property list`.
+.. Execute the `internal` request by invoking the `<entity-collection-source/>` for `destination entity set` with `$filter=referenced property` in `(...source property list)`.
+
+=== Error Handling and Error Propagation
+
+Error handling for OData v4 applications is local to each flow. Whether you use `generic expansion` or not. Error handling code for a given `EntitySet` will belong to the flow implementations for that entity set, both for normal requests and for the internal requests observed during expansion processing.
+
+Handling errors during the expansion process is not supported, because the expansion process runs once the execution of the flow is finished. There are two options available for handling errors:
+
+- *Ignore errors during expansion:* Errors during generic expansion will be ignored. +
+Expansion of collections and single objects will be resolved to empty lists and `null` values respectively (default behaviour).
+- *Propagate errors during expansion:* Errors during generic expansion will arise. +
+When an error reaches the OData v4 router, the original request will be answered with an error response.
+
+Error handling is configured *per-source* on the *Capabilities* tab.
+
+=== Limitations
+
+* Only the first `ReferencialConstraint` is considered. +
+Navigation properties that use compound keys are not supported.
+* Expansion is done `field-by-field` instead of `EntitySet-by-EntitySet`. +
+This may cause more internal requests than necessary.
+* The expanded entities should always have an `<entity-collection-source/>` implementation. +
+Expansion using `<entity-source/>` implementations are not executed due to performance concerns.
+
+== Implementation Guidelines
+
+To implement the `generic expansion` feature, each source has to be annotated with the relevant types that might be expanded from it. Note that enabling `$expand` in a completely unrestricted way will enable API clients to write arbitrarily complex queries, which can create performance issues.
```

Comment:
```suggestion
To implement the `generic expansion` feature, you must annotate each source must with the types that it can expand. Enabling `$expand` without restrictions enables API clients to write arbitrarily complex queries, which can create performance issues.
```

Reply:
OK! Thanks!